helix-term/src/ui/editor.rs

```diff
@@ -530,19 +530,33 @@ impl EditorView {
                                 (grapheme.as_ref(), width)
                             };
 
+                            let style = if is_whitespace {
+                                style.patch(whitespace_style)
+                            } else {
+                                style
+                            };
```

Comment:
Can we inline this into both branches? I don't mind the duplication.

This is a very hot loop that currently gets ran for *all* characters on the line, including the ones past the right edge of the screen. We'd waste time calculating the style there.